src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -2164,7 +2170,7 @@ void InterpreterSelectQuery::executeFetchColumns(QueryProcessingStage::Enum proc
      *  instead of max_threads, max_distributed_connections is used.
      */
     bool is_remote = false;
-    if (storage && storage->isRemote())
+    if (storage && storage->isRemote(context))
     {
         is_remote = true;
         max_threads_execute_query = max_streams = settings.max_distributed_connections;
```

Comment:
This is relevant to parallel replicas.
Interesting thing is that `Distributed` ignores max_streams in `read` method (so probably `max_distributed_connections` is useless?), and this is only used for `merge_threads` value at AggregatingStep (and I don't understand why not max_threads). It looks like we can make `max_streams` local.